samples/durable-task-sdks/java/monitoring/src/main/java/io/durabletask/samples/MonitoringPattern.java

```diff
@@ -0,0 +1,199 @@
+                        state.getRuntimeStatus() == OrchestrationRuntimeStatus.FAILED ||
+                        state.getRuntimeStatus() == OrchestrationRuntimeStatus.TERMINATED) {
+                        
+                        logger.info("Monitoring completed with status: " + state.getRuntimeStatus());
+                        
+                        if (state.getRuntimeStatus() == OrchestrationRuntimeStatus.COMPLETED) {
+                            JobResult result = state.readOutputAs(JobResult.class);
+                            logger.info("Final result: " + result.status);
+                            logger.info("Message: " + result.message);
+                        } else if (state.getRuntimeStatus() == OrchestrationRuntimeStatus.FAILED) {
+                            logger.warning("Monitoring failed: " + state.getFailureDetails());
+                        }
+                        
+                        break;
+                    }
+                }
+                
+                // Wait before checking again
+                Thread.sleep(waitInterval.toMillis());
+                totalWaitTime = totalWaitTime.plus(waitInterval);
+            }
+            
+            if (totalWaitTime.compareTo(maxWaitTime) >= 0) {
+                logger.warning("Client timed out waiting for orchestration to complete");
+            }
+            
+        } finally {
+            worker.stop();
+        }
+    }
+
+    private static DurableTaskGrpcWorker createWorker(String connectionString) {
+        return DurableTaskSchedulerWorkerExtensions.createWorkerBuilder(connectionString)
+            .addOrchestration(new TaskOrchestrationFactory() {
+                @Override
+                public String getName() {
+                    return ORCHESTRATION_NAME;
+                }
+
+                @Override
+                public TaskOrchestration create() {
+                    return ctx -> {
+                        // Get the job data
+                        JobData jobData = ctx.getInput(JobData.class);
+                        
+                        // Set initial status
+                        ctx.setCustomStatus(new JobStatus("Starting monitoring..."));
+                        
+                        // Create a timeout timer
+                        Task timeoutTask = ctx.createTimer(Duration.ofSeconds(jobData.timeoutSeconds));
+                        
+                        // Simulate job monitoring with status updates
+                        int pollingCount = 0;
+                        while (true) {
+                            // Update status
+                            ctx.setCustomStatus(new JobStatus("Polling job status (attempt " + (++pollingCount) + ")"));
+                            
+                            // Simulate some work
+                            ctx.createTimer(Duration.ofSeconds(jobData.pollingIntervalSeconds));
```

Comment:
This line is missing `.await()`. It won't block the orchestration without it.
```suggestion
                            ctx.createTimer(Duration.ofSeconds(jobData.pollingIntervalSeconds)).await();
```